core/templates/pages/exploration-editor-page/exploration-save-and-publish-buttons/exploration-save-and-publish-buttons.component.spec.ts

```diff
@@ -84,6 +127,19 @@ describe('Exploration save and publish buttons component', () => {
            provide: ExternalSaveService,
            useClass: MockExternalSaveService
          },
+         {
+           provide: WindowRef,
+           useClass: MockWindowRef
+         },
+         {
+           provide: ContextService,
+           useValue: {
+             getExplorationId: () => {
+               return 'exp1';
+             },
+             setExplorationIsLinkedToStory: () => {}
```

Comment:
Can we spy on ContextService instead of mocking them?

Reply:
done

Reply:
50 runs: https://github.com/heyimShivam/frontend-tests-flake-check-oppia/actions/runs/3270779983/jobs/5379810281

Reply:
@heyimShivam I don't think the errors are completely resolved yet: https://github.com/gp201/frontend-tests-flake-check-oppia/actions/runs/3270885486/jobs/5380031374#step:10:1696

Reply:
I have runned it twice all test are passing are you sure you have last tested commit